modules/java/Makefile.am

```diff
@@ -43,9 +43,18 @@ $(SYSLOG_NG_CORE_JAR): $(JAVA_CLASS_FILES) $(SYSLOG_NG_CORE_LIBS_DIR)
 $(SYSLOG_NG_CORE_LIBS_DIR) $(SYSLOG_NG_CORE_CLASSES_DIR) $(SYSLOG_NG_CORE_PACKAGE_DIR) $(JAVA_BUILDDIR):
 	mkdir -p $@
 
+if HAVE_JAVAH
+
 $(JAVA_HEADER_FILES): $(JAVA_BUILDDIR)/org_syslog_ng_%.h: $(SYSLOG_NG_CORE_PACKAGE_DIR)/%.class
 	$(AM_V_GEN)$(JAVAH) -classpath  $(SYSLOG_NG_CORE_CLASSES_DIR) -d $(JAVA_BUILDDIR) org.syslog_ng.$*
 
+else
+
+JAVAC_HEADER_OPTIONS = -h $(JAVA_BUILDDIR)
+$(JAVA_HEADER_FILES): $(JAVA_CLASS_FILES)
```

Comment:
The `JAVA_CLASS_FILES` target will generate our headers as a side effect. I can't express this explicitly in the makefile because a [static pattern](https://www.gnu.org/software/make/manual/html_node/Static-Usage.html#Static-Usage) is used in `JAVA_CLASS_FILES`.